lib/index.js

```diff
@@ -10,23 +10,41 @@ const Joi = require('joi');
 const Schema = require('./schema');
 const Package = require('../package.json');
 
-// const SchwiftyMigration = require('schwifty-migration');
-
 const internals = {};
 
 exports.register = function (server, options, next) {
 
+    Joi.assert(options, Schema.plugin, 'Bad plugin options passed to schwifty.');
+
     const rootState = internals.state(server.root);
 
-    internals.state(server).dir = options.dir;
     const knexFile = options.knexFile;
 
-    internals.state(server).knex = Knex(knexFile[process.env.NODE_ENV]);
+    /*
+        Move the models into the knexFile object
+        Since a models path isn't part of it's schema on knexjs.org
+    */
+
+    if (options.models) {
+        knexFile.models = options.models;
+    }
+
+    let knex;
```

Comment:
What if different models (say, in different plugins) are on different DBs?  I only ask because (with the help of waterline) this was a feature of dogwater.  Maybe it's not something we're still interested in?